website/docs/reference/sources/file.md.erb

```diff
@@ -96,6 +112,107 @@ be [autodiscovered](#auto-discovery) continually at a rate defined by the
 
 Each line is read until a new line delimiter (the `0xA` byte) or `EOF` is found.
 
+###  Multi-Line Messages
+
+Sometimes a single log event will appear as multiple log lines. To handle this,
+Vector provides a set of `multiline` options. These options were carefully
+thought through and will allow you to solve the simplest and most complex
+cases. Let's look at a few examples:
+
+#### Example 1: Ruby Exceptions
+
+Ruby exceptions, when logged, consist of multiple lines:
+
+```
+foobar.rb:6:in `/': divided by 0 (ZeroDivisionError)
+	from foobar.rb:6:in `bar'
+  from foobar.rb:2:in `foo'
+  from foobar.rb:9:in `<main>'
+```
+
+To consume these lines as a single event with Vector use the following
+configuration:
+
+```toml
+[sources.my_file_source]
+  type = "file"
+  # ...
+
+  [sources.my_file_source.multiline]
+    start_pattern = "^[^\\s]"
+    mode = "continue_through"
+    condition_pattern = "^[\\s]+from"
+```
+
+* `start_pattern`, set to `^[^\\s]`, tells Vector that the start of new events
+  should _not_ start  with white-space.
+* `mode`, set to `continue_through`, tells Vector continue aggregating lines
+  until the `condition_pattern` is no longer valid (exluding the invalid line).
+* `condition_pattern`, set to `^[\\s]+from`, tells Vector to continue
+  aggregating lines if they start with white-space followed by `from`.
+
+#### Example 2: Line Continuations
+
+Some programming languages use the backslash (`\`) character to signal that a
+line will continue on the next line:
+
+```
+This is a line that is\
+continued to the next line
+```
+
+To consume these lines as a single event with Vector use the following
+configuration:
+
+```toml
+[sources.my_file_source]
+  type = "file"
+  # ...
+
+  [sources.my_file_source.multiline]
+    start_pattern = "^[^\\s]"
+    mode = "continue_past"
+    condition_pattern = "\\$"
```

Comment:
@MOZGIII is this example correct?

Reply:
In our test cases we have a different example:
https://github.com/timberio/vector/blob/a00295e59ca789f9c1f100a99f6f06c5c772bb84/src/sources/file/line_agg.rs#L291-L320

What you have would still work, but I think we better switch to the sample from our tests.